src/daemon/daemon_main.cpp

```diff
@@ -84,6 +84,11 @@ int main_impl(int argc, char* argv[])
 
     mp::daemon::monitor_and_quit_on_settings_change(); // TODO replace with async restart in relevant settings handlers
     mp::Daemon daemon(std::move(config));
+    QObject::connect(&app,
+                     &QCoreApplication::aboutToQuit,
+                     &daemon,
+                     &mp::Daemon::shutdown_grpc_server,
```

Comment:
For my understanding, why not connect to `daemon_rpc` directly?

Reply:
Because we don't have access to it from here. It is a private member of `Daemon`. We could expose a getter to it, and then we could connect to it directly from here. Would that be preferable?

Reply:
Oh duh, I read this as if it was in `daemon.cpp`. Nevermind